### PR TITLE
Click a recording to select it

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/4647
* Closes https://github.com/rerun-io/rerun/issues/4649

You can now click a recording to select it (show info about it in the selection panel).

select-recording

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4761/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4761/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4761/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4761)
- [Docs preview](https://rerun.io/preview/482599187dd686bfa0e1c4709ed090d6e84c3c9d/docs) 
- [Examples preview](https://rerun.io/preview/482599187dd686bfa0e1c4709ed090d6e84c3c9d/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by emilk** [comment]
Blocked on having a "Store" icon

File: crates/re_data_ui/src/item.rs [hunk 1]
```diff
@@ -1,26 +0,0 @@
-use re_viewer_context::{Item, UiVerbosity, ViewerContext};
-
-use crate::DataUi;
-
-impl DataUi for Item {
```
Comment (Wumpf) [inline crates/re_data_ui/src/item.rs:5]: do we know what this was used for?

File: crates/re_viewer/src/app.rs [hunk 2]
```diff
@@ -400,6 +400,14 @@ impl App {
                     );
                 }
             }
+
+            SystemCommand::SetSelection(store_id, item) => {
```
Comment (Wumpf) [inline crates/re_viewer/src/app.rs:404]: This makes me wonder if we should move more over to this system command. If everything where to use it we wouldn't need double buffering on the selection state 🤔

**Review by Wumpf — APPROVED** [review]
really cool, I like the collateral code cleanups almost more than the feature itself :)

File: crates/re_data_ui/src/item.rs (hunk 1 above)
Comment (emilk) [reply crates/re_data_ui/src/item.rs:5]: Selection panel and time panel; but not anymore

File: crates/re_viewer/src/app.rs (hunk 2 above)
Comment (emilk) [reply crates/re_viewer/src/app.rs:404]: Yup, I think we should

**Pull request merged**